### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by mosquito**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+       print(instance)
+
+       main_instance = await MainClass()
+       related_instance = await RelatedClass(main_instance)
+
+       assert not main_instance.is_closed
+       assert not related_instance.is_closed
+
+       await main_instance.close()
+       assert not main_instance.is_closed
```
Comment (Alviner) [inline README.rst]: why parent is not closed here if you wait close on the line above?

File: async_class.py [hunk 2]
```diff
@@ -135,31 +162,71 @@ def create_future(self) -> asyncio.Future:
...
-            return f
+            return
 
-        self.__closed = True
+        tasks: List[Union[asyncio.Future, Coroutine]] = []
+
+        if hasattr(self, "_async_class_task_store"):
+            tasks.append(self.__adel__())
+            tasks.append(self.__tasks__.close(exc))
+            self.__closed = True
```
Comment (Alviner) [inline async_class.py:194]: but here in close u switch closed flag. Code is reachable cuz you call await instance and task store exists here

**Review by Alviner — COMMENTED** [review]
Can we discuss a little about linking?

File: README.rst (hunk 1 above)
Comment (mosquito) [reply README.rst]: Bug I guess

**Review by Alviner — APPROVED** [review]
LGTM

**Comment by Alviner** [comment]
🥇

**Review by Alviner — APPROVED** [review]

**Review by Alviner — APPROVED** [review]

**Pull request merged**